correlation_modules/3a_validate_module_composition.py

```diff
@@ -0,0 +1,463 @@
+            )
+
+            # array_mapping[rnaseq_module] = p-value
+            array_mapping[array_group_name] = pval
+
+        # Find the module with the best mapping (i.e. lowest p-value)
+        best_array_module = min(array_mapping, key=array_mapping.get)
+
+        rows.append(
+            {
+                "RNA-seq module id": rnaseq_group_name,
+                "best array module id": best_array_module,
+                "p-value": min(array_mapping.values()),
+            }
+        )
+
+    rnaseq_to_array_df = pd.DataFrame(rows)
+
+    return rnaseq_to_array_df
+
+
+# %%time
+rnaseq_to_array_true = map_rnaseq_to_array_modules(
+    pao1_rnaseq_membership_df, pao1_array_membership_df
+)
+
+rnaseq_to_array_true.head()
+
+# %%time
+rnaseq_to_array_shuffle = map_rnaseq_to_array_modules(
+    pao1_rnaseq_membership_df, pao1_array_membership_shuffle_df
+)
+
+rnaseq_to_array_shuffle.head()
+
+# ## Plot
+
+# +
+# Format df for plotting
+array_to_rnaseq_true["label"] = "true"
+array_to_rnaseq_shuffle["label"] = "shuffle"
+
+array_to_rnaseq_combined = pd.concat([array_to_rnaseq_true, array_to_rnaseq_shuffle])
+
+# +
+rnaseq_to_array_true["label"] = "true"
+rnaseq_to_array_shuffle["label"] = "shuffle"
+
+rnaseq_to_array_combined = pd.concat([rnaseq_to_array_true, rnaseq_to_array_shuffle])
+# -
+
+array_to_rnaseq_combined.head()
+
+# +
+# Test: mean p-values using the true module labels vs shuffle module labels is significant
+true_array_to_rnaseq = array_to_rnaseq_combined[
+    array_to_rnaseq_combined["label"] == "true"
+]["p-value"].values
+shuffled_array_to_rnaseq = array_to_rnaseq_combined[
+    array_to_rnaseq_combined["label"] == "shuffle"
+]["p-value"].values
+
+(stats, pvalue) = scipy.stats.ttest_ind(true_array_to_rnaseq, shuffled_array_to_rnaseq)
+print(pvalue)
+# -
+
+f = sns.boxplot(
+    x=array_to_rnaseq_combined["label"],
+    y=array_to_rnaseq_combined["p-value"],
+    palette="Blues",
+    notch=True,
+)
+f.set_title("Array to RNA-seq modules")
+
+# +
+# Test: mean p-values using the true module labels vs shuffle module labels is significant
+true_rnaseq_to_array = rnaseq_to_array_combined[
+    rnaseq_to_array_combined["label"] == "true"
+]["p-value"].values
+shuffled_rnaseq_to_array = rnaseq_to_array_combined[
+    rnaseq_to_array_combined["label"] == "shuffle"
+]["p-value"].values
+
+(stats, pvalue) = scipy.stats.ttest_ind(true_rnaseq_to_array, shuffled_rnaseq_to_array)
+print(pvalue)
+# -
+
+g = sns.boxplot(
+    x=rnaseq_to_array_combined["label"],
+    y=rnaseq_to_array_combined["p-value"],
+    palette="Blues",
+    notch=True,
+)
+g.set_title("RNA-seq to array modules")
+
+# **Takeaway:**
+#
+# * Comparing how RNA-seq modules map to array modules, looks like most modules have very low p-values, meaning they map well between RNA-seq and array.
+# * In contrast, random modules have slightly higher p-values, indicating that the modules don't map as well, though the p-values are still very low.
```

Comment:
Why do the random modules have low p-values? Is it because genes' expression is highly correlated?

Reply:
Very good catch! After correcting for multiple tests, I get trends that are more expected (i.e. low corrected p-values for true labels and high corrected p-values for shuffled labels). However, if I examine the overlap between modules, all of the best mapping based on corrected p-value scores, are 0. This behavior is unexpected to me - I wonder if this test is in appropriate given the rarety of the event. From my reading, I think I've implemented this test correctly. I made a note in this notebook about this and I'll have to return to this. This analysis of validating that our modules are capturing relevant biology was more critical when we were planning to do analyses on these modules but with our new story, it looks like this step isn't too necessary.